packages/lesswrong/components/posts/TableOfContents/TableOfContentsRow.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { PureComponent, Component } from 'react';
+import { Components, registerComponent } from 'meteor/vulcan:core';
+import { withStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import classNames from 'classnames';
+
+const styles = theme => ({
+  root: {
+    ...theme.typography.body2,
+    ...theme.typography.postStyle,
+  },
+
+  // For the highlighted section only, disable the half-opacity-on-hover effect
+  // that's otherwise globally applied to <a> tags.
+  highlighted: {
+    fontWeight: "bold",
+    '& $link': {
+      color: "black !important",
+    },
+    "& a:focus, & a:hover": {
+      opacity: "initial",
+    }
+  },
+
+  link: {
+    display: "block",
+    whiteSpace: "nowrap",
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    paddingTop: 4,
+    paddingBottom: 4,
+    color: theme.palette.grey[700],
+    '&:hover':{
+      opacity:1,
+      fontWeight:"bold",
+    }
+  },
+
+  level0: {
+    display:"inline-block",
+    maxWidth:240,
+    '&:first-of-type': {
+      marginBottom: theme.spacing.unit,
+      paddingBottom: theme.spacing.unit,
+      borderBottom: "solid 1px rgba(0,0,0,.2)",
+    },
+    '&:last-of-type': {
+      marginTop: theme.spacing.unit,
+      paddingTop: theme.spacing.unit,
+      borderTop: "solid 1px rgba(0,0,0,.2)",
+    }
+  },
+  level1: {
+    paddingLeft: 0,
+  },
+  level2: {
+    fontSize:"1.05em",
+    paddingLeft: 16,
+    "& $link": {
+      color: "rgba(0,0,0,.5)",
+    }
+  },
+  level3: {
+    fontSize:"1.05em",
+    color:theme.palette.grey[700],
+    paddingLeft: 32,
+    "& $link": {
+      color: "rgba(0,0,0,.5)",
+    }
+  },
+  level4: {
+    fontSize:"1.05em",
+    color:theme.palette.grey[700],
+    paddingLeft: 48,
+    "& $link": {
+      color: "rgba(0,0,0,.5)",
+    }
+  },
+});
+
+class TableOfContentsRow extends PureComponent
+{
+  render() {
+    const {indentLevel=0, highlighted=false, href, onClick, children, classes} = this.props;
+
+    return <Typography variant="body2"
+      className={classNames(
+        classes.root,
+        this.levelToClassName(indentLevel),
+        { [classes.highlighted]: highlighted }
+      )}
+    >
+      <a href={href} onClick={onClick} className={classes.link}>
+        {children}
+      </a>
+    </Typography>
+  }
+
+  levelToClassName = (level) => {
```

Comment:
You have this function twice, maybe we want to abstract it

Reply:
Agreed, although not 100% sure where it should live.